src/gc.c

```diff
@@ -1300,11 +1283,19 @@ STATIC_INLINE jl_value_t *jl_gc_pool_alloc_inner(jl_ptls_t ptls, int pool_offset
         if (v != NULL) {
             // like the freelist case,
             // but only update the page metadata when it is full
-            jl_gc_pagemeta_t *pg = jl_assume(page_metadata((char*)v - 1));
+            jl_gc_pagemeta_t *pg = jl_assume(page_metadata_unsafe((char*)v - 1));
             assert(pg->osize == p->osize);
             pg->nfree = 0;
             pg->has_young = 1;
-            v = *(jl_taggedvalue_t**)cur_page;
+            pg = pop_page_metadata_back(&ptls->page_metadata_lazily_freed);
+            if (pg != NULL) {
+                v = gc_reset_page(ptls, p, pg);
+                pg->osize = p->osize;
+                push_page_metadata_back(&ptls->page_metadata_allocd, pg);
+            }
+            else {
+                v = NULL;
```

Comment:
v is NULL (you just branched on that)

Reply:
Quite the opposite, right? E.g. we enter this code block if `v != NULL` (line 1283) and if `pg == NULL` (line 1291). The `if` block in 1291 is the only one which could re-assign `v`.

Reply:
ah, I miscounted the indent on this one